packages/near-lake-primitives/src/types/block.ts

```diff
@@ -0,0 +1,159 @@
+    }
+
+    receipts(): Array<Receipt> {
+        if (this.executedReceipts.length == 0) {
+            this.executedReceipts = this.streamerMessage.shards.flatMap((shard) => shard.receiptExecutionOutcomes).map((executionReceipt) => outcomeWithReceiptToReceipt(executionReceipt))
+        }
+        return this.executedReceipts;
+    }
+
+    actions(): Action[] {
+        const actions: Action[] = this.streamerMessage.shards.flatMap((shard) => shard.receiptExecutionOutcomes).filter((exeuctionOutcomeWithReceipt) => isActionReceipt(exeuctionOutcomeWithReceipt.receipt)).map((exeuctionOutcomeWithReceipt) => receiptViewToAction(exeuctionOutcomeWithReceipt.receipt)).filter((action) => action !== null).map(action => action as Action)
+        return actions
+    }
+
+    events(): Event[] {
+        const events = this.receipts().flatMap((executedReceipt) => executedReceipt.logs.map(logToRawEvent).map((rawEvent) => {
+            if (rawEvent) {
+                let event = { relatedReceiptId: executedReceipt.receiptId, rawEvent: rawEvent } as Event
+                return event
+            } else {
+                return null
+            }
+        })).filter((event) => event !== null).map((event) => event as Event)
+        return events
+    }
+    stateChanges(): StateChange[] {
+        if (this._stateChanges.length == 0) {
+            this._stateChanges = this.streamerMessage.shards.flatMap((shard) => shard.stateChanges).map(fromStateChangeViewToStateChange)
+        }
+        return this._stateChanges
+    }
+    actionByReceiptId(receipt_id: string): Action | undefined {
+        if (this._actions.size == 0) {
+            this.buildActionsHashmap()
+        }
+        return this._actions.get(receipt_id);
+    }
+
+    eventsByReceiptId(receipt_id: string): Array<Event> {
+        if (this._events.size == 0) {
+            this.buildEventsHashmap()
+        }
+        return this._events.get(receipt_id) || [];
+    }
+
+    eventsByAccountId(account_id: string): Array<Event> {
+        return this.events().filter((event) => {
+            const action = this.actionByReceiptId(event.relatedReceiptId)
+            if (action !== undefined && action?.receiverId == account_id || action?.signerId == account_id) {
+                return true
+            }
+            return false
+        });
+    }
+
+    private buildActionsHashmap() {
+        const actions = new Map<string, Action>();
+        this.actions().forEach(action => {
+            actions.set(action.receiptId, action)
+        }
+        );
+        this._actions = actions
+    }
+
+    private buildEventsHashmap(): Map<string, Array<Event>> {
+        const events = new Map<string, Array<Event>>();
+        for (const receipt of this.executedReceipts) {
+            events.set(receipt.receiptId, receipt.events);
+        }
+        return events;
+    }
+
+
+}
+
+export function createBlock(streamerMessage: StreamerMessage): Block {
```

Comment:
I would rather use the naming you've used for conversion `streamerMessageToBlockHeader`. Here we could use `streamerMessageToBlock` as `createBlock` doesn't make sense, we're not creating it actually.